File: mobile/flutter_app/lib/transport/transport.dart
import 'dart:async';
import 'dart:typed_data';

enum TransportType {
  wifi,
}

abstract class Transport {
  TransportType get type;
  bool get isConnected;

  Future<void> connect();
  Future<void> disconnect();
  Future<void> send(Uint8List data);

  Stream<Uint8List> get onData;
  Stream<void> get onClose;
  Stream<dynamic> get onError;
}

File: agent/src/transport/transport.ts
export enum TransportType {
  WIFI = 'wifi',
}

export interface Transport {
  readonly type: TransportType;
  deviceId: string;

  connect(): Promise<void>;
  disconnect(): Promise<void>;

  send(data: Buffer | string): Promise<void>;

  onData(handler: (data: Buffer | string) => void): void;
  onClose(handler: () => void): void;
  onError(handler: (err: Error) => void): void;

  isConnected(): boolean;
}
